lib/galaxy/tools/recommendations.py

```diff
@@ -100,13 +100,32 @@ def __set_model(self, trans, remote_model_url):
             for k in tool_pos_sorted:
                 self.tool_weights_sorted[k] = tool_weights[str(k)]
             # collect ids and names of all the installed tools
+            all_t = dict()
             for tool_id, tool in trans.app.toolbox.tools():
                 t_id_renamed = tool_id
                 if t_id_renamed.find("/") > -1:
                     t_id_renamed = t_id_renamed.split("/")[-2]
-                self.all_tools[t_id_renamed] = (tool_id, tool.name)
+                if t_id_renamed not in all_t:
+                    all_t[t_id_renamed] = list()
+                all_t[t_id_renamed].append((tool_id, tool.version_object, tool.name))
+            self.all_tools = self.__get_latest_tool(all_t)
         return True
 
+    def __get_latest_tool(self, a_tools):
+        """
+        Sort tool versions and take the latest one
+        """
+        all_tools = dict()
+        for t in a_tools:
+            t_versions = list()
+            for t_v in a_tools[t]:
```

Comment:
Would be better to get the index without sorting. 

Something along the lines of:
```
max_index, max_value = max(enumerate(values), key=operator.itemgetter(1))
```

Reply:
thanks, will test it and update